### PR TITLE
Issue #14759: Enhanced cmdline options documentation by tabular format

### LINKED ISSUE #14759: Enhance Command Line documentation by organizing options in Tabular format
Extension of PR #14710 and response to https://github.com/checkstyle/checkstyle/pull/14710#discussion_r1550049366

Current cmdline docs : https://checkstyle.org/cmdline.html#Command_line_usage
Our command line options are bulleted under one section at present.

A better way of documenting for end users would be organizing all the available cmdline options in a Tabular format.
The table would consist of 2 columns (Titles can be "`Command Line Option`" and "`Description`" or decided further) 
and 18 rows for each of the 18 options that we host.

This format of documentation is the standard and can be found followed by Apache maven CLI options too
https://maven.apache.org/ref/3.6.1/maven-embedder/cli.html#

### PR DESCRIPTION
**Pull request by MANISH-K-07**
Aims to close #14759 

## WIP ....

### CODE REVIEW

**Comment by MANISH-K-07** [comment]
Github, generate website

**Comment by MANISH-K-07** [comment]
Github, generate website

**Comment by MANISH-K-07** [comment]
Github, generate website

**Comment by MANISH-K-07** [comment]
The anchors are working just as suggested at https://github.com/checkstyle/checkstyle/issues/14759#issuecomment-2038627651

Not yet sure why the test is unable to read option -c description from doc. ([failure](https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=21060&view=logs&j=c902ebb4-c9f8-5f09-4e17-ff78fbbc842e&t=9ca98c81-ff64-58f0-9d03-a23ac1c4a111&l=1397))
Looking into it....

@romani , @rnveach , @nrmancuso , any thoughts on this ?

**Comment by nrmancuso** [comment]
> @romani , @rnveach , @nrmancuso , any thoughts on this ?

The table looks great, let's make CI happy

**Comment by MANISH-K-07** [comment]
> let's make CI happy

Still trying to debug why the `-c option` node is not being read despite the fact that it is available in the table :(

**Comment by MANISH-K-07** [comment]
@romani , @nrmancuso , 
I am having a bit of trouble debugging the test file. Could you please help me with why the firstChild is not being mapped?

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/CliOptionsXdocsSyncTest.java [hunk 1]
```diff
@@ -49,7 +49,8 @@ public void validateCliDocSections() throws Exception {
         final Map<String, String> cmdDesc = new HashMap<>();
 
         final NodeList sections = getSectionsFromXdoc("src/xdocs/cmdline.xml.vm");
-        final Set<String> cmdOptions = getListById(sections.item(2), "CLI_Options");
+        final Set<String> cmdOptions =
+                getListById(sections.item(2), "Command_line_usage_Command_line_options");
```
Comment (nrmancuso) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/CliOptionsXdocsSyncTest.java:53]: What is 2 here?

Comment (MANISH-K-07) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/CliOptionsXdocsSyncTest.java:53]: @nrmancuso , 2 is the section index of the XML (considering 0-based), I believe.

0. Content
1. Description 
2. **Command Line Usage**

Comment (MANISH-K-07) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/CliOptionsXdocsSyncTest.java:53]: @nrmancuso , could you please assign this PR to yourself.

Comment (nrmancuso) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/CliOptionsXdocsSyncTest.java:53]: > @nrmancuso , could you please assign this PR to yourself.

No need to assign reviewers until CI is green. CI is the first reviewer :)

Comment (MANISH-K-07) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/CliOptionsXdocsSyncTest.java:53]: Got it @nrmancuso , please see https://github.com/checkstyle/checkstyle/pull/14799#discussion_r1571327442.

I seem to be falling short on what I'm missing while debug 😕 
Theoretically, the siblings are read only after firstChild is mapped and so on, but not sure why this is being skipped and continued to next node.

**Comment by romani** [comment]
@MANISH-K-07 , please help to finish this PR

**Comment by MANISH-K-07** [comment]
> @MANISH-K-07 , please help to finish this PR

https://github.com/checkstyle/checkstyle/pull/14799#discussion_r1571624993 still no thought on this..

**Comment by romani** [comment]
>[ERROR]   CliOptionsXdocsSyncTest.validateCliDocSections:75 CLI Option: -c present in Main.java but not documented in cmdline.xml.vm
expected not to be: null

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/CliOptionsXdocsSyncTest.java (hunk 1 above)
Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/CliOptionsXdocsSyncTest.java:53]: This PR changes xml a lot, but test that does selection from this xml is not changed. `getSectionsFromXdoc(` should be changed a bit.

Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/CliOptionsXdocsSyncTest.java:53]: It use to be:
```
<ul id="CLI_Options">
        <li>
          <code>-c
```
Becomes a way more complicated and different:
```
....
<tbody>
              <tr>
                <td>
                  <a name="c"/>
                  <a href="#c">
                    <strong>-c</strong> configurationFile
                  </a>
```

So getListById should be rewritten.
@MANISH-K-07 , do need help to write a code?

Comment (MANISH-K-07) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/CliOptionsXdocsSyncTest.java:53]: Will look into it at the earliest @romani

**Comment by romani** [comment]
GitHub, generate website

**Comment by romani** [comment]
It would be awesome to put anchor to each line as we did at this table https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/6b63421_2024160342/google_style.html#a1

It would help a lot to reference exact parameter for users when they have a question

**Comment by rnveach** [comment]
Option color is the anchor in this iteration. There is no image, it is just a clickable link.

**Comment by romani** [comment]
Ok, this works, we just need to fix test to pass.

**Pull request closed without merging**